flytekit/models/sagemaker/training_job.py

```diff
@@ -0,0 +1,231 @@
+from __future__ import absolute_import
+
+from typing import List
+from flyteidl.plugins.sagemaker import training_job_pb2 as _training_job_pb2
+from flytekit.models import common as _common
+
+
+class TrainingJobResourceConfig(_common.FlyteIdlEntity):
+    def __init__(
+            self,
+            instance_count: int,
+            instance_type: str,
+            volume_size_in_gb: int,
+    ):
+        self._instance_count = instance_count
+        self._instance_type = instance_type
+        self._volume_size_in_gb = volume_size_in_gb
+
+    @property
+    def instance_count(self) -> int:
+        """
+        :rtype: int
+        """
+        return self._instance_count
+
+    @property
+    def instance_type(self) -> str:
+        """
+        :rtype: str
+        """
+        return self._instance_type
+
+    @property
+    def volume_size_in_gb(self) -> int:
+        """
+        :rtype: int
+        """
+        return self._volume_size_in_gb
+
+    def to_flyte_idl(self) -> _training_job_pb2.TrainingJobResourceConfig:
+        """
+
+        :rtype: _training_job_pb2.TrainingJobResourceConfig
+        """
+        return _training_job_pb2.TrainingJobResourceConfig(
+            instance_count=self.instance_count,
+            instance_type=self.instance_type,
+            volume_size_in_gb=self.volume_size_in_gb,
+        )
+
+    @classmethod
+    def from_flyte_idl(cls, pb2_object: _training_job_pb2.TrainingJobResourceConfig):
+        """
+
+        :param pb2_object:
+        :rtype: TrainingJobResourceConfig
+        """
+        return cls(
+            instance_count=pb2_object.instance_count,
+            instance_type=pb2_object.instance_type,
+            volume_size_in_gb=pb2_object.volume_size_in_gb,
+        )
+
+
+class MetricDefinition(_common.FlyteIdlEntity):
+    def __init__(
+            self,
+            name: str,
+            regex: str,
+    ):
+        self._name = name
+        self._regex = regex
+
+    @property
+    def name(self) -> str:
+        """
+
+        :rtype: str
+        """
+        return self._name
+
+    @property
+    def regex(self) -> str:
```

Comment:
we might need to add documentation here :(. This is the sad part of models. You can do that maybe?